packages/integration-sdk-runtime/src/logger/index.ts

```diff
@@ -279,10 +300,10 @@ export class IntegrationLogger extends EventEmitter
       `Error occurred while validating integration configuration.`,
     );
 
-    if (isUserConfigError(err)) {
-      this.warn({ errorId, err }, description);
-    } else {
+    if (shouldReportErrorToOperator(err)) {
       this.error({ errorId, err }, description);
+    } else {
+      this.warn({ errorId, err }, description);
```

Comment:
I'm a little uneasy about the impression this code gives, that `.error()` is somehow going to report to the operator. We used to have it that way, when there was a metric on `level: 50` log events that would trigger an alert and PD notification, but it isn't this way now.

The [code that calls this `validationFailure()` function](https://github.com/JupiterOne/sdk/blob/77a669a26de908cac6040a5ef480da2d622cd747/packages/integration-sdk-runtime/src/execution/executeIntegration.ts#L101) re-throws the error, which means that the code catching that also needs to know whether the error is meant to be reported to an operator.

I'm thinking we should consider a path toward avoiding throwing errors out of the SDK which do not need to be reported to the operator. The managed integration runtime should be able to report everything thrown out of the sdk.

What do you think @austinkelleher @ndowmon ?

Reply:
I agree that the managed integration runtime should be able to report everything thrown out of the SDK